packages/legend-application-query/src/components/QueryBuilderPostFilterPanel.tsx

```diff
@@ -421,9 +425,20 @@ const QueryBuilderPostFilterConditionEditor = observer(
     // Drag and Drop on filter condition value
     const handleDrop = useCallback(
       (item: QueryBuilderParameterDragSource): void => {
-        node.condition.setValue(item.variable.parameter);
+        if (
+          isTypeCompatibleWithConditionValueType(
+            item.variable.parameter.genericType?.value.rawType,
+            guaranteeNonNullable(node.condition.columnState.getReturnType()),
```

Comment:
What does this mean for derived projeciton columns. 
We should probably escape the check for those if not known. I wouldn't do a `guaranteeNonNullable`

Reply:
getReturnType is an abstract function that will be overrode by `QueryBuilderDerivationProjectionColumnState` and `QueryBuilderSimpleProjectionColumnState`.